test/jdk/javax/swing/JButton/SwingButtonResizeTestWithOpenGL.java

```diff
@@ -0,0 +1,328 @@
+            public void focusGained(FocusEvent fe) {
+                focusGainedLatch.countDown();
+            }
+        });
+        frame.getContentPane().setLayout(new FlowLayout());
+        frame.getContentPane().add(button);
+        frame.pack();
+        frame.setVisible(true);
+        frame.toFront();
+    }
+
+    public static void main(String[] args) throws Exception {
+        focusGainedLatch = new CountDownLatch(1);
+        SwingButtonResizeTestWithOpenGL test = new SwingButtonResizeTestWithOpenGL();
+        test.runTest();
+    }
+
+    public void runTest() throws Exception {
+        BufferedImage bimage1;
+        BufferedImage bimage2;
+
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+            robot.setAutoDelay(200);
+
+            if (focusGainedLatch.await(3, TimeUnit.SECONDS)) {
+                System.out.println("Button focus gained...");
+            } else {
+                System.out.println("Button focus not gained...");
+                throw new RuntimeException("Can't gain focus on button even after waiting too long..");
+            }
+
+            System.out.println("Getting initial button image..image1");
+            bimage1 = getButtonImage();
+
+            File file1 = new File("image1.png");
+            saveButtonImage(bimage1, file1);
+
+            // some platforms may not support maximize frame
+            if (frame.getToolkit().isFrameStateSupported(JFrame.MAXIMIZED_BOTH)) {
+                robot.waitForIdle();
+                //maximize frame from normal size
+                frame.setExtendedState(JFrame.MAXIMIZED_BOTH);
+                System.out.println("Frame is maximized");
+                robot.waitForIdle();
+
+                if (frame.getToolkit().isFrameStateSupported(JFrame.NORMAL)) {
+                    System.out.println("Frame is back to normal");
+                    //resize from maximum size to normal
+                    frame.setExtendedState(JFrame.NORMAL);
+
+                    // capture image of JButton after resize
+                    System.out.println("Getting image of JButton after resize..image2");
+                    bimage2 = getButtonImage();
+                    File file2 = new File("image2.png");
+                    saveButtonImage(bimage2, file2);
+
+                    // compare button images from before and after frame resize
+                    DiffImage di = new DiffImage(bimage1.getWidth(), bimage1.getHeight());
+                    di.compare(bimage1, bimage2);
+                    System.out.println("Taking the diff of two images, image1 and image2");
+                    // results of image comparison
+                    int numDiffPixels = di.getNumDiffPixels();
+                    if (numDiffPixels != 0) {
+                        throw new RuntimeException("Button renderings are different after window resize, num of Diff Pixels=" + numDiffPixels);
```

Comment:
I think the `compare` method can be modified to return `true/false` based on the nDiff value. If  **nDiff is not equal to zero** which you are checking at [L300](https://github.com/openjdk/jdk/blob/5694b6811e06793ddaf1984e909cc0e925d93047/test/jdk/javax/swing/JButton/SwingButtonResizeTestWithOpenGL.java#L300) then you can return `false`, evaluate the return value inside if condition and throw RTE with the num of diff pixels value. If **nDiff is equal to zero** then compare method should return `true` and test should pass.

Code snippet can be changed to something like this...
                    

> System.out.println("Taking the diff of two images, image1 and image2");
>                     // results of image comparison
>                     if (!di.compare(bimage1, bimage2)) {
>                         throw new RuntimeException("Button renderings are different after window resize, num of Diff Pixels=" + di.getNumDiffPixels());